File: backend-java/src/main/java/com/asmolabs/zanshin/repository/entities/SSHKey.java
package com.asmolabs.zanshin.repository.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "ssh_key")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SSHKey {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(name = "private_key", columnDefinition = "TEXT", nullable = false)
    private String privateKey;

    @Column(name = "public_key", columnDefinition = "TEXT")
    private String publicKey;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "sshKey")
    private List<ZanshinRepository> repositories;
}

File: zanshin/models/ssh_key.py
import uuid
from zanshin.clock import utcnow
from sqlalchemy import Column, String, Text
from zanshin.database import Base
from zanshin.models.guid import GUID
from zanshin.models.safedatetime import SafeDateTime

class SSHKey(Base):
    __tablename__ = "ssh_key"

    id = Column(GUID, primary_key=True, default=uuid.uuid4)
    name = Column(String(255), nullable=False)
    private_key = Column(Text, nullable=False)
    public_key = Column(Text, nullable=True)
    created_at = Column(SafeDateTime, default=utcnow, nullable=False)
